spec/code-snippets/DeclarationReferences.ts

```diff
@@ -0,0 +1,443 @@
+ */
+export class Class15 {
+  /**
+   * Shortest name:  {@link Class15."abc. def"}
+   * Full name:      {@link Class15[class]."abc. def"[static]}
+   */
+  public static 'abc. def': string = 'static member with malformed characters';
+
+  /**
+   * Shortest name:  {@link Class15."abc. def"}
+   * Full name:      {@link Class15[class]."abc. def"[instance]}
+   */
+  public 'abc. def': string = 'instance member with malformed characters';
+
+  /**
+   * Shortest name:  {@link Class15.static}
+   * Also valid:     {@link Class15."static"}
+   * Full name:      {@link Class15[class].static[static]}
+   */
+  public static static: string = 'static member using keyword as name';
+
+  /**
+   * Shortest name:  {@link Class15."\uD842\uDFB7"}
+   * Full name:      {@link Class15[class]."\uD842\uDFB7"[instance]}
+   *
+   * NOTE: The string in double quotes is parsed using JSON.parse(), which converts
+   * this surrogate pair expression to the corresponding Unicode character.
+   */
+  public '𠮷': string = 'instance member using JSON unicode escapes';
+
+  /**
+   * Shortest name:  {@link Class15."\\\""}
+   * Full name:      {@link Class15[class]."\\\""[instance]}
+   *
+   * Again, the string `"\\\""` is passed to JSON.parse().
+   */
+  public '\\"': string = 'instance member using JSON escapes';
+
+  /**
+   * Shortest name:  {@link Class15.\{\}}
+   * JSON approach:  {@link Class15."\u007B\u007D"}
+   * Full name:      {@link Class15[class].\{\}[instance]}
+   *
+   * NOTE: The closing curly brace is problematic because it is a TSDoc inline tag delimiter.
+   * The "shortest name" solves this problem using TSDoc backslash escapes.  The "JSON approach"
+   * instead uses JSON unicode escapes.  Both are valid, although the JSON approach is more likely
+   * to be compatible with documentation parsers that are not TSDoc compliant.
+   */
+  public '{}': string = 'instance member using TSDoc delimiters';
+
+  /**
+   * Shortest name:  {@link Class15."1.5"}
+   * Full name:      {@link Class15[class]."1.5"[instance]}
+   *
+   * Note that the actual JavaScript object key will become a string, so "1.5" is a correct
+   * way to reference this item.
+   */
+  public 1.5: string = "a number as the key";
+}
+
+//---------------------------------------------------------
+// Generic parameters are not part of the notation
+
+/**
+ * Shortest name:  {@link Type16}
+ * Full name:      {@link Type16[type]}
+ *
+ * Note that "<T>" is never part of the declaration reference notation.
+ * In the TypeScript language, signatures cannot be distinguished by generic parameters.
+ */
+export type Type16<T> = T | Error;
+
+//---------------------------------------------------------
+// Operators must be selected using explicit labels
+
+/**
+ * Shortest name:  {@link Interface17}
+ * Full name:      {@link Interface17[interface]}
+ */
+export interface Interface17 {
+  /**
+   * Shortest name:  {@link Interface17.operator[STRING_INDEXER]}
```

Comment:
>operator [](start = 40, length = 8)

this won't work if you have a member called "operator"

What if you could say `{@link Interface17.[STRING_INDEXER]}`